src/Workspaces/Core/Portable/Shared/Extensions/ICompilationExtensions.cs

```diff
@@ -1,11 +1,47 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.Linq;
 
 namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static class ICompilationExtensions
     {
+        public static ImmutableArray<Compilation> GetReferencedCompilations(this Compilation compilation)
+        {
+            var builder = ArrayBuilder<Compilation>.GetInstance();
```

Comment:
To be honest, my main takeaway was that only very specialized code should consume `Compilation.References`.  Even the features that didn't need to change were better served by the list of referenced assemblies on the source module.

Furthermore, this particular method is probably not a good candidate for publication because it explicitly drops module references.  I don't believe that would be appropriate for a compiler API.

If you like, you can file a bug asking for a rationalization of these APIs.  @tmat might have more thoughts.